common/models/user.js

```diff
@@ -993,18 +995,22 @@ module.exports = function(User) {
 
   User.validatePassword = function(plain) {
     var err;
-    if (plain && typeof plain === 'string' && plain.length <= MAX_PASSWORD_LENGTH) {
-      return true;
+    if (!plain || typeof plain !== 'string') {
+      err = new Error(g.f('Invalid password.'));
+      err.code = 'INVALID_PASSWORD';
```

Comment:
That validation is not only for newly registered users, but for resetting passwords as well, so we can be pointing a potential hackers to to where they should dig  deeper :)